tools/gomod-local-update/internal/updater/updater.go

```diff
@@ -0,0 +1,268 @@
+	}
+	sha := strings.Split(string(out), "\t")[0]
+	if len(sha) == 0 {
+		return "", time.Time{}, fmt.Errorf("%w: empty SHA from git ls-remote", ErrModOperation)
+	}
+
+	// Validate the SHA
+	if err := u.validateSHA(sha); err != nil {
+		return "", time.Time{}, err
+	}
+
+	// Use u.git.Command for show
+	out, err = u.git.Command(ctx, "show", "-s", "--format=%cI", sha)
+	if err != nil {
+		return "", time.Time{}, fmt.Errorf("failed to get commit time: %w", err)
+	}
+
+	commitTime, err := time.Parse(gitTimeFormat, strings.TrimSpace(string(out)))
+	if err != nil {
+		return "", time.Time{}, fmt.Errorf("failed to parse commit time: %w", err)
+	}
+
+	return sha[:gitSHALength], commitTime, nil
+}
+
+// Run starts the module update process
+func (u *Updater) Run() error {
+	logger := log.New(os.Stdout, "", log.LstdFlags)
+
+	logger.Printf("info: auto-detecting modules with local replace directives")
+
+	f, err := u.readModFile()
+	if err != nil {
+		return fmt.Errorf("%w: failed to read and parse go.mod file", ErrModOperation)
+	}
+
+	// Auto-detect modules to update
+	modulesToUpdate, err := u.findLocalReplaceModules()
+	if err != nil {
+		return fmt.Errorf("%w: failed to detect local replace modules", ErrModOperation)
+	}
+	if len(modulesToUpdate) == 0 {
+		logger.Printf("info: no modules found to update in %s", f.Module.Mod.Path)
+		return nil
+	}
+	logger.Printf("info: found %d modules with local replace directives: %v",
+		len(modulesToUpdate), modulesToUpdate)
+
+	// Get commit info once for all modules
+	sha, commitTime, err := u.getGitInfo(u.config.RepoRemote, u.config.BranchTrunk)
+	if err != nil {
+		if errors.Is(err, ErrInvalidConfig) {
+			return err
+		}
+		return fmt.Errorf("%w: failed to get git commit info from remote", ErrModOperation)
+	}
+
+	// Update the modules in the same file handle
+	if err := u.updateGoMod(f, modulesToUpdate, sha, commitTime); err != nil {
+		return fmt.Errorf("%w: failed to update module versions in go.mod", ErrModOperation)
+	}
+
+	return u.writeModFile(f)
+}
+
+// updateGoMod updates the go.mod file with new pseudo-versions
+func (u *Updater) updateGoMod(modFile *modfile.File, modulesToUpdate []string, sha string, commitTime time.Time) error {
+	for _, modulePath := range modulesToUpdate {
+		moduleExists := false
+		majorVersion := getMajorVersion(modulePath)
+		pseudoVersion := module.PseudoVersion(majorVersion, "", commitTime, sha[:gitSHALength])
+
+		// Find and update version
+		for _, req := range modFile.Require {
+			if req.Mod.Path == modulePath {
+				moduleExists = true
+				if u.config.DryRun {
+					log.Printf("[DRY RUN] Would update %s: %s => %s", modulePath, req.Mod.Version, pseudoVersion)
+					continue
+				}
+
+				if err := modFile.AddRequire(modulePath, pseudoVersion); err != nil {
+					return fmt.Errorf("%w: failed to update version for module %s",
+						ErrModOperation, modulePath)
+				}
+				break
+			}
+		}
+
+		if !moduleExists {
+			continue
+		}
```

Comment:
Seems like this is the last statement in the for loop anyways, so is this continue doing anything? And is this variable doing anything?

Reply:
Good catch, I think this was a remnant from a refactor. Removed in 1d9497a42ab9874ac932b286659e90d6594df1d8.